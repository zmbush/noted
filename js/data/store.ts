// Copyright 2021 Zachary Bush.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.
//
import { AnyAction, combineReducers, configureStore } from '@reduxjs/toolkit';

import notes from 'data/notes/slice';
import ui from 'data/ui/slice';
import user from 'data/user/slice';

export const appReducer = combineReducers({ user, notes, ui });

export const rootReducer = (state: ReturnType<typeof appReducer>, action: AnyAction) => {
  if (action.type === 'RESET') {
    return appReducer(undefined, action);
  }
  return appReducer(state, action);
};
export const store = configureStore({
  reducer: rootReducer,
  devTools: true,
});

export type AppState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
