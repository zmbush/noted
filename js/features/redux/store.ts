// Copyright 2021 Zachary Bush.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.
//
import { combineReducers, configureStore } from '@reduxjs/toolkit';

import errorTracking from 'features/error_tracking/slice';
import noteLoading from 'features/note_loading/slice';
import noteEdit from 'features/notes/edit/slice';
import notes from 'features/notes/list/slice';
import requestTracking from 'features/request_tracking/slice';
import user from 'features/user/slice';

export const rootReducer = combineReducers({
  errorTracking,
  noteEdit,
  noteLoading,
  notes,
  requestTracking,
  user,
});
export const createStore = () =>
  configureStore({
    reducer: rootReducer,
    devTools: true,
  });

export const store = createStore();

export type AppState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
