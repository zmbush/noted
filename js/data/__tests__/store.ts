// Copyright 2019 Zachary Bush.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.
import { rootReducer } from '../store';

describe('data::reducers()', () => {
  test('matches snapshot', () => {
    expect(rootReducer(undefined, { type: null })).toMatchInlineSnapshot(`
      Object {
        "notes": Object {
          "entities": Object {},
          "ids": Array [],
        },
        "ui": Object {
          "inProgress": Object {},
          "lastError": Object {
            "any": null,
          },
          "noteChanging": Object {},
        },
        "user": Object {
          "isSignedIn": false,
          "user": null,
        },
      }
    `);
  });
});
