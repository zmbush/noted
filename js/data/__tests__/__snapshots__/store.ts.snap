// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data::reducers() matches snapshot 1`] = `
Object {
  "notes": Object {
    "entities": Object {},
    "ids": Array [],
  },
  "ui": Object {
    "inProgress": Array [],
    "lastError": null,
  },
  "user": Object {
    "isSignedIn": false,
    "user": null,
  },
}
`;
